### PR TITLE
(enhance)charts: add support for namespaced mode of operation

### PR DESCRIPTION
**Pull request by ksatchit**
Signed-off-by: ksatchit <karthik.s@mayadata.io>

#### What this PR does / why we need it:

- Modifies the templates to support namespaced mode of operation (refer: https://docs.litmuschaos.io/docs/next/namespaced-mode/) 
- Adds a new label to the resources `litmuschaos.io/version` pointing to the Chart.AppVersion, i.e., litmus release

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #https://github.com/litmuschaos/litmus/issues/1799

#### Special notes for your reviewer:

- The `Notes` file has been modified to provide consistent steps/details for all the three supported operational modes (keeping in mind the user performing a namespaced install mostly won't have the permissions to list CRDs or clusterroles etc.,)

   - Existing of API resources implies CRDs are present, and thereby the step to get CRDs has been removed
   - A successful bring up of the chaos pod suggests the creation of the serviceaccount w/ the roles, so the step to get clusterroles/clusterrolebindings is removed. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] DCO signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: charts/litmus/values.yaml [hunk 1]
```diff
@@ -56,9 +56,8 @@ tolerations: []
 
 affinity: {}
 
-# Support for https://docs.litmuschaos.io/docs/admin-mode
-administratorMode:
-  enabled: false
+# Supports: standard, admin, namespaced
+operatorMode: namespaced
```
Comment (ispeakc0de) [inline charts/litmus/values.yaml:60]: What is the default operatorMode? standard or namespaced?

Comment (ksatchit) [reply charts/litmus/values.yaml]: It will be standard @ispeakc0de !! Just pushed a commit. Thanks for noticing

**Comment by ksatchit** [comment]
#### Tests via `helm template` for the various modes of installation 

- [standard.txt](https://github.com/litmuschaos/litmus-helm/files/5056887/standard.txt)
- [namespaced.txt](https://github.com/litmuschaos/litmus-helm/files/5056888/namespaced.txt)
- [admin.txt](https://github.com/litmuschaos/litmus-helm/files/5056890/admin.txt)

File: charts/litmus/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "1.6.2"
 description: A Helm chart to install litmus infra components on Kubernetes
 name: litmus
-version: 1.6.2
+version: 1.6.3
```
Comment (uditgaurav) [inline charts/litmus/Chart.yaml:5]: It shouldn't sync with litmus version?

Comment (ksatchit) [reply charts/litmus/Chart.yaml:5]: The chart versions are expected to be independently bumped whenever they go undergo any change, irrespective of whether it is a release change or not. The litmus release is indicated by the AppVersion

**Comment by ksatchit** [comment]
#### Tests for chaos execution in different modes

- standard mode

```yaml
root@test:~# kubectl describe chaosresult 
Name:         nginx-chaos-pod-delete
Namespace:    default
Labels:       name=nginx-chaos-pod-delete
Annotations:  <none>
API Version:  litmuschaos.io/v1alpha1
Kind:         ChaosResult
Metadata:
  Creation Timestamp:  2020-08-11T12:43:54Z
  Generation:          2
  Managed Fields:
    API Version:  litmuschaos.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        f:experimentstatus:
          f:failStep:
          f:phase:
          f:verdict:
    Manager:         pod-delete
    Operation:       Update
    Time:            2020-08-11T12:44:25Z
  Resource Version:  476755
  Self Link:         /apis/litmuschaos.io/v1alpha1/namespaces/default/chaosresults/nginx-chaos-pod-delete
  UID:               8833b4d4-ece5-4882-a084-9541e4a8debd
Spec:
  Engine:      nginx-chaos
  Experiment:  pod-delete
Status:
  Experimentstatus:
    Fail Step:  N/A
    Phase:      Completed
    Verdict:    Pass
Events:
  Type    Reason   Age   From                     Message
  ----    ------   ----  ----                     -------
  Normal  Summary  43s   pod-delete-pshs38-4qqqz  pod-delete experiment has been Passed
```

File: charts/litmus/Chart.yaml (hunk 2 above)
Comment (Jasstkn) [reply charts/litmus/Chart.yaml:5]: @uditgaurav no, we sync litmus version with appVersion. Chart version changes every time when we push changes into master.

**Comment by ksatchit** [comment]
#### Tests for multiple/parallel chaos execution via several namespaced operators acting at once (along w/ presence of admin operator)

**Comment by ksatchit** [comment]
@Jasstkn @ispeakc0de @uditgaurav , could you perform a final review, I have made some slight modifications to the templates.

File: charts/litmus/Chart.yaml (hunk 2 above)
Comment (Jasstkn) [reply charts/litmus/Chart.yaml:5]: It's important for all of the users. They can rely on each version of helm-chart as a final and immutable package.

**Pull request merged**